### PR TITLE
Solution to JS Reference Type Diagnostic

### PR DESCRIPTION
**Pull request by glasneed**

### CODE REVIEW

**Comment by glasneed** [comment]
Comfort: 3
Clarity: 3

I used  my class notes to help with this diagnostic. However, I couldn't complete the last function due to time. Still unclear on the unique function.

File: lib/diagnostic.js [hunk 1]
```diff
@@ -1,34 +1,70 @@
...
-const uniqueWords = function (/* */) {
+const uniqueWords = function (word) {
   /* your response here */
+   let normalizedWords = normalizeWords(text)
+
+   //Have to pseudo code as I didn't know how to loop through and get this value.
+   //I would create a for loop that loops through normalizedWords
+   //and then apply logic to test if it is unique. However, I'm
+   //not clear on how to do this and will try over weekend
+
```
Comment (benjimelito) [inline lib/diagnostic.js:38]: Looks like sound logic here, let me know if you don't end up figuring it out with more time.

File: lib/diagnostic.js [hunk 2]
```diff
@@ -1,34 +1,70 @@
...
+  let uniqueWordsFound = uniqueWords (words)
+  theNumberOfWords = uniqueWordsFound.length
+
+} else if (uniqueOption = false) {
+    let normalizedWordsFound = normalizeWords(text)
+    theNumberOfWords = normalizedWordsFound.length
+  }
+
+  return theNumberOfWords;
+
```
Comment (benjimelito) [inline lib/diagnostic.js:61]: Assuming that you had the above function working, I think this would do what you want it to.

**Comment by benjimelito** [comment]
Like I said in the code, let me (or someone else) know if you don't end up getting a working solution with more time. 2/4

**Pull request closed without merging**